install/kubernetes/helm/istio/values.yaml

```diff
@@ -229,6 +229,14 @@ global:
       host: # example: statsd-svc.istio-system
       port: # example: 9125
 
+    # Sets the Envoy Metrics Service address, used to push envoy metrics to an external collector
+    # via the Metrics Service gRPC API.
+    # Disabled by default.
+    envoyMetricsService:
+      enabled: false
+      host: # example: metrics-service.istio-system
+      port: # example: 15000
+
```

Comment:
Could you combine this and the envoyStatsD into one single struct like this:

```
# These are statistics corresponding to the inner workings of the envoy proxies in the data plane.
# They contain detailed stat information emitted by filters in envoy as well as other subsystems in Envoy,
# gathered across multiple requests and connections. This should not be confused with Istio telemetry 
# service that generates stats (e.g., response codes, latencies, etc.) per request or connection.
# Note that in addition to the following two sinks where Envoy pushes stats, prometheus style metrics can 
# be scraped from the envoys using the admin port 127.0.0.1:15000, or via 127.0.0.1:15090/stats/prometheus
# Disabled by default
envoyInternalStats:
 statsdSink
   enabled: ..
   ...
  grpcStatsReceiverSink:
   enabled: ...
```

and fix up the existing statsD references as well?

The reason we need to do this is because the term metrics service is utterly generic, and ends up conflating with Istio metrics/telemetry. this will cause user confusion when they want to enable/disable pieces of Istio. 

On a similar note, could you please rename the API field in istio/api to envoy_internal_stats_grpc_sink ?

cc @douglas-reid - this sound good?

Reply:
@rshriram yes. this sounds reasonable to me. thanks for taking on this bit of work.

Reply:
@rshriram - The reasoning behind that merger with the statsd config makes sense, but that would also be a breaking change for helm template users that modify those values. Is that an issue? I'm fine with making the change, just want to make sure we're not breaking any expected backwards compatibility guarantees.

Reply:
hmm true.. in that case lets keep them as two separate pieces but add the big blob on top.and probably name the metrics service thing as envoyInternalStatsGrpcReceiver for the lack of a more uglier lengthier name..

Reply:
dont forget to update the comment above